py/server/deephaven/jcompat.py

```diff
@@ -24,7 +24,7 @@ def j_array_list(values: Iterable = None) -> jpy.JType:
         return None
     r = jpy.get_type("java.util.ArrayList")(len(list(values)))
     for v in values:
-        r.add(v)
+        r.add(unwrap(v))
```

Comment:
Good change. Consider whether the map conversions need to unwrap keys. Right now they unwrap values, but not keys, and this seems necessary but not sufficient. Should probably be a follow-up issue/PR.